### PR TITLE
rename actor/remote/slf4j/serialization modules

### PR DESCRIPTION
**Pull request by pjfanning**
Rename some modules. Aim is to cover other modules like discovery, persistence and streams in a later PR.

Part of https://github.com/apache/incubator-pekko/issues/21

### CODE REVIEW

**Comment by danischroeter** [comment]
How do we want to tackle the client transition from akka to pekko?
We could start by adding the necessary _client adaptions_ when we perform client relevant changes like this pr?
I fear that if we do not start to track this while _breaking_ the existing clients it will get even harder...
We could add a _AkkaToPekkoMigration.md_ in the root and modify this along with the relevant PR's... wdyt?

**Comment by pjfanning** [comment]
Thanks @danischroeter

This PR is already quite large and it will take another 2 or 3 just to rename the rest of the modules in this one repo.

The net effect is that when we start publishing jars, they will be called pekko-xyz instead of akka-xyz. We will not be able to continue to call the jars akka-xyz because Lightbend have the trademark.

**Comment by danischroeter** [comment]
> Thanks @danischroeter
> 
> This PR is already quite large and it will take another 2 or 3 just to rename the rest of the modules in this one repo.
> 
> The net effect is that when we start publishing jars, they will be called pekko-xyz instead of akka-xyz. We will not be able to continue to call the jars akka-xyz because Lightbend have the trademark.

Jep that's all clear and good. Already if we link these _client breaking changes_ to #22 we'll get minimal tracking.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Akka [![Latest version](https://index.scala-lang.org/akka/akka/akka-actor/latest.svg)](https://index.scala-lang.org/akka/akka/akka-actor)[![Build Status](https://api.travis-ci.com/akka/akka.svg?branch=main)](https://travis-ci.com/github/akka/akka)
+Akka [![Latest version](https://index.scala-lang.org/akka/akka/pekko-actor/latest.svg)](https://index.scala-lang.org/akka/akka/pekko-actor)[![Build Status](https://api.travis-ci.com/akka/akka.svg?branch=main)](https://travis-ci.com/github/akka/akka)
```
Comment (jrudolph) [inline README.md:1]: A bit weird to fix links half-way...

File: build.sbt [hunk 2]
```diff
@@ -453,30 +453,30 @@ lazy val streamTestsTck = akkaModule("akka-stream-tests-tck")
...
   .dependsOn(actor, slf4j)
-  .settings(AutomaticModuleName.settings("akka.actor.typed"))
+  .settings(AutomaticModuleName.settings("pekko.actor.typed"))
   .settings(Dependencies.actorTyped)
   .settings(OSGi.actorTyped)
   .settings(initialCommands :=
     """
-      import akka.actor.typed._
-      import akka.actor.typed.scaladsl.Behaviors
+      import pekko.actor.typed._
```
Comment (jrudolph) [inline build.sbt:470]: Was that intentional?

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 3]
```diff
@@ -841,7 +841,8 @@ public void usePatternsAskPipe() {
         // using timeout from above
         CompletableFuture<Object> future2 = ask(actorB, "another request", t).toCompletableFuture();
 
-        CompletableFuture<Result> transformed = future1.thenCombine(future2, (x, s) -> new Result((String) x, (String) s));
+        CompletableFuture<Result> transformed =
```
Comment (jrudolph) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java:844]: How did these reformats happen?

**Review by jrudolph — CHANGES_REQUESTED** [review]
Reviewed all the non-documentation files, mostly LGTM.

There's the package change in build.sbt which seems unintentional (and potentially wrong based on what the final package name will be). => Request Changes

In general, I would much prefer to break these PRs down in pieces of functionality instead of doing plain search/replaces on everything, e.g. documentation fixes outside of links to files are not critical right now and could be changed and reviewed later.

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java (hunk 3 above)
Comment (pjfanning) [reply akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: I reverted this

File: build.sbt (hunk 2 above)
Comment (pjfanning) [reply build.sbt]: I reverted this

**Comment by pjfanning** [comment]
@jrudolph I made a few changes based on your review - I'll probably need to make more changes later today - and I'll rebase to lower the number of commits

**Comment by pjfanning** [comment]
@jrudolph I looked at the changes again and don't see anything obviously wrong. When you have time, could you have a look again?

**Comment by nvollmar** [comment]
@pjfanning The doc generation issue seems related to https://github.com/lightbend/paradox/issues/491
Updating to the latest version of sbt-paradox-akka fixes it: 
```
addSbtPlugin("com.lightbend.akka" % "sbt-paradox-akka" % "0.45")
```
We're going to need to create our own version of it to style the documentation.

Fixing this yields a host of doc generation issues due to the rename of modules:

```
 Error writing content: project-info.conf does not contain `akka-actor` at /Users/nvo/Development/git/incubator-pekko/akka-docs/src/main/paradox/actors.md
[error] Could not find file for snippet at /Users/nvo/Development/git/incubator-pekko/akka-docs/src/main/paradox/coordinated-shutdown.md:21
[error] :   @@snip [reference.conf](/akka-actor/src/main/resources/reference.conf) { #coordinated-shutdown-phases }
[error]     ^
[error] Could not find file for snippet at /Users/nvo/Development/git/incubator-pekko/akka-docs/src/main/paradox/event-bus.md:8
[error] :  @@snip [EventBus.scala](/akka-actor/src/main/scala/akka/event/EventBus.scala) { #event-bus-api }
[error]    ^
[error] Could not find file for snippet at /Users/nvo/Development/git/incubator-pekko/akka-docs/src/main/paradox/general/configuration-reference.md:18
[error] @@snip [reference.conf](/akka-actor/src/main/resources/reference.conf)
[error] ^
[error] Could not find file for snippet at /Users/nvo/Development/git/incubator-pekko/akka-docs/src/main/paradox/general/configuration-reference.md:23
[error] @@snip [reference.conf](/akka-actor-typed/src/main/resources/reference.conf)
[error] ^
```

File: akka-stream-tests/src/test/scala/akka/stream/io/DeprecatedTlsSpec.scala [hunk 4]
```diff
@@ -516,7 +516,7 @@ class DeprecatedTlsSpec extends StreamSpec(DeprecatedTlsSpec.configOverrides) wi
 
         Source.single(SendBytes(ByteString.empty)).via(flow).runWith(Sink.ignore)
       }
-      Await.result(run("akka-remote"), 3.seconds) // CN=akka-remote
+      Await.result(run("pekko-remote"), 3.seconds) // CN=pekko-remote
```
Comment (nvollmar) [inline akka-stream-tests/src/test/scala/akka/stream/io/DeprecatedTlsSpec.scala:519]: The CN in the trust store is `akka-remote`, probably best to keep it akka-remote for now and open a separate issue to create new keystore/truststore for pekko.

File: akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala [hunk 5]
```diff
@@ -564,7 +564,7 @@ class TlsSpec extends StreamSpec(TlsSpec.configOverrides) with WithLogCapturing
           Source.single(SendBytes(ByteString.empty)).via(flow).runWith(Sink.ignore)
         }
 
-        Await.result(run("akka-remote"), 3.seconds) // CN=akka-remote
+        Await.result(run("pekko-remote"), 3.seconds) // CN=pekko-remote
```
Comment (nvollmar) [inline akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala:567]: Same issue here

File: akka-stream-tests/src/test/scala/akka/stream/io/DeprecatedTlsSpec.scala (hunk 4 above)
Comment (pjfanning) [reply akka-stream-tests/src/test/scala/akka/stream/io/DeprecatedTlsSpec.scala]: I've reverted the changes in these 2 tests

File: akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala (hunk 5 above)
Comment (pjfanning) [reply akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala]: reverted

**Comment by pjfanning** [comment]
@nvollmar I updated the paradox plugin and tried to fix up the akka/pekko renames in akka-docs module. akka-docs now builds for me locally.

**Review by nvollmar — APPROVED** [review]
lgtm

**Comment by pjfanning** [comment]
current status is that the CI build appears to have a hanging test in pekko-remote

**Comment by pjfanning** [comment]
I fixed the broken pekko-remote test

**Comment by Claudenw** [comment]
Shouldn't the directory names be changed as part of this change?

**Comment by pjfanning** [comment]
> Shouldn't the directory names be changed as part of this change?

All the akka-actor*/akka-remote*/akka-slf4j*/akka-serialization* dirs are renamed in this PR. Maybe Github doesn't make that obvious.

The vast majority of the files modified in this PR are just modified due to the dir moves - that there are no actual text changes in those files.

I'm avoiding renaming all the dirs in one PR to avoid having super-massive PRs.

**Comment by pjfanning** [comment]
@jrudolph would you be able to re-review this when you get a chance?

**Comment by pjfanning** [comment]
this PR is badly affected by the recent package name merge - I will abandon it

**Pull request closed without merging**